core/src/main/java/org/elasticsearch/index/indexing/IndexingSlowLog.java

```diff
@@ -42,6 +43,12 @@
     private long indexInfoThreshold;
     private long indexDebugThreshold;
     private long indexTraceThreshold;
+    /**
+     * How much of the source to log in the slowlog - 0 means log none and
+     * anything greater than 0 means log at least that many <em>characters</em>
+     * of the source.
+     */
+    private int sourceToLog;
```

Comment:
also maybe a more descriptive name could be useful, maybe something like `maxSourceCharsToLog`?